### PR TITLE
ATO-171: Action interventions after IPV

### PR DESCRIPTION
**Pull request by ethanmills**
## What?

Action account interventions post-IPV. Best reviewed commit-wise

### CODE REVIEW

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java [hunk 1]
```diff
@@ -90,12 +90,12 @@ public void incrementLogout(
             Optional<AccountInterventionStatus> accountInterventionStatus) {
         String accountInterventionStr = "unknown";
         if (accountInterventionStatus.isPresent()) {
-            if (accountInterventionStatus.get().blocked()) {
-                accountInterventionStr = "blocked";
-            }
             if (accountInterventionStatus.get().suspended()) {
                 accountInterventionStr = "suspended";
             }
+            if (accountInterventionStatus.get().blocked()) {
```
Comment (billrdunn) [inline orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java:96]: Good spot! Would it be better to use a more descriptive string eg. `SUSPENDED_NO_ACTION`?

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/LogoutService.java [hunk 2]
```diff
@@ -69,11 +63,11 @@ public LogoutService(
     public APIGatewayProxyResponseEvent handleAccountInterventionLogout(
             Session session,
             APIGatewayProxyRequestEvent input,
-            Optional<String> clientId,
-            Optional<String> sessionId,
+            String clientId,
```
Comment (billrdunn) [inline orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/LogoutService.java:66]: Going to be some conflicts with my ATO-342 PR due to this change

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java [hunk 3]
```diff
@@ -146,7 +160,17 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
...
                     processingStatus);
             if (processingStatus == ProcessingIdentityStatus.COMPLETED) {
-                checkAccountInterventionService(pairwiseSubject.getValue(), auditContext);
+                var aisResult =
+                        segmentedFunctionCall(
+                                "AIS: getAccountStatus",
+                                () ->
+                                        accountInterventionService.getAccountStatus(
+                                                internalPairwiseSubjectId, auditContext));
+                if (configurationService.isAccountInterventionServiceActionEnabled()
```
Comment (billrdunn) [inline ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java:169]: Move this flag to within `logoutService.handleAccountInterventionLogout`? To match the pattern in `getAccountStatus`

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java [hunk 4]
```diff
@@ -233,10 +248,16 @@ public APIGatewayProxyResponseEvent handleRequest(
...
-                        ipvCallbackHelper.getAccountInterventionStatus(
-                                internalPairwiseSubjectId, auditContext);
-                if (configurationService.isAccountInterventionServiceActionEnabled()) {
-                    ipvCallbackHelper.doAccountIntervention(accountInterventionStatus);
+                        segmentedFunctionCall(
+                                "AIS: getAccountStatus",
+                                () ->
+                                        this.accountInterventionService.getAccountStatus(
+                                                internalPairwiseSubjectId, auditContext));
+                if (configurationService.isAccountInterventionServiceActionEnabled()
```
Comment (billrdunn) [inline ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java:256]: Move this flag to within `logoutService.handleAccountInterventionLogout` to match the pattern above?

**Review by billrdunn — COMMENTED** [review]
Squash the second and third commit together?

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java (hunk 3 above)
Comment (billrdunn) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java:169]: Can also apply this to `AuthenticationCallbackHandler` once ATO-342 is merged

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java (hunk 1 above)
Comment (ethanmills) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java:96]: I'm not sure what you mean?

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java (hunk 3 above)
Comment (ethanmills) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/ProcessingIdentityHandler.java:169]: I think the return makes that a little tricky?

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java (hunk 1 above)
Comment (billrdunn) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/CloudwatchMetricsService.java:96]: Will resolve this as it's not blocking approval of PR

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**